### PR TITLE
Fix imports to avoid multiple inclusions

### PR DESCRIPTION
**Pull request by jander-msft**
Overview of changes:

- Unify imports for MicroBuild to avoid multiple inclusion.
- Unify imports for managed projects for SDK and non-SDK style projects.
- Exclude all assets from MicroBuild.Core package to prevent NuGet from automatically importing props and targets.
- Remove superfluous Microsoft.Cpp.props imports.

### CODE REVIEW

File: build/MicroBuild.props [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(PackagesDir)\Microsoft.VisualStudioEng.MicroBuild.Core\$(MicroBuildCoreVersion)\build\Microsoft.VisualStudioEng.MicroBuild.Core.props" Condition="Exists('$(PackagesDir)\Microsoft.VisualStudioEng.MicroBuild.Core\$(MicroBuildCoreVersion)\build\Microsoft.VisualStudioEng.MicroBuild.Core.props')" />
```
Comment (WilliamXieMSFT) [inline build/MicroBuild.props:5]: >Condition [](start = 156, length = 9)

nit move this to newline

File: src/Extensions.Base.Api/Extensions.Base.Api.csproj [hunk 2]
```diff
@@ -2,8 +2,10 @@
 <!-- Copyright (c) Microsoft Corporation. All rights reserved.
      Licensed under the MIT License. -->
 <Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), EnlistmentRoot.marker))\build\Common.props" />
-  <Import Project="$(EnlistmentRoot)\build\Managed.props" />
+  <PropertyGroup>
+    <UseDotNetSdk>false</UseDotNetSdk>
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), EnlistmentRoot.marker))\build\Managed.props" />
```
Comment (WilliamXieMSFT) [inline src/Extensions.Base.Api/Extensions.Base.Api.csproj:8]: >$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), EnlistmentRoot.marker)) [](start = 19, length = 93)

$(EnlistmentRoot) ?

File: src/Extensions.Base.Api.Tests/Extensions.Base.Api.Tests.csproj [hunk 3]
```diff
@@ -2,8 +2,10 @@
 <!-- Copyright (c) Microsoft Corporation. All rights reserved.
      Licensed under the MIT License. -->
 <Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), EnlistmentRoot.marker))\build\Common.props" />
-  <Import Project="$(EnlistmentRoot)\build\Managed.props" />
+  <PropertyGroup>
+    <UseDotNetSdk>false</UseDotNetSdk>
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), EnlistmentRoot.marker))\build\Managed.props" />
```
Comment (WilliamXieMSFT) [inline src/Extensions.Base.Api.Tests/Extensions.Base.Api.Tests.csproj:8]: >$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), EnlistmentRoot.marker)) [](start = 19, length = 93)

$(EnlistmentRoot) ?

**Review by WilliamXieMSFT — APPROVED** [review]
:shipit:

**Review by delmyers — APPROVED** [review]

File: src/Extensions.Base.Api/Extensions.Base.Api.csproj (hunk 2 above)
Comment (jander-msft) [reply src/Extensions.Base.Api/Extensions.Base.Api.csproj:8]: This is the first import of the project so the EnlistmentRoot property hasn't been set yet.

File: src/Extensions.Base.Api.Tests/Extensions.Base.Api.Tests.csproj (hunk 3 above)
Comment (jander-msft) [reply src/Extensions.Base.Api.Tests/Extensions.Base.Api.Tests.csproj:8]: This is the first import of the project so the EnlistmentRoot property hasn't been set yet.

**Pull request merged**